internal/aws/awsutil/conn.go

```diff
@@ -369,8 +332,18 @@ func getSTSRegionalEndpoint(r string) string {
 	return e
 }
 
-func GetDefaultSession(logger *zap.Logger, options session.Options) (*session.Session, error) {
-	result, serr := session.NewSessionWithOptions(options)
+func GetDefaultSession(logger *zap.Logger, cfg *AWSSessionSettings) (*session.Session, error) {
+	awsConfig := &aws.Config{}
+	cCredentialProvider := customCredentialProvider(cfg)
+	for i := 0; i < len(cCredentialProvider); i++ {
+		retrieve, err := cCredentialProvider[i].Retrieve()
```

Comment:
This is a bit different from the CWA implementation. In CWA, we don't retrieve the actual credentials when evaluating the chain. All it takes for a credential provider to get selected is for the condition check to pass. https://github.com/aws/amazon-cloudwatch-agent/blob/main/cfg/aws/credentials.go#L61

Reply:
True. But does doing it this way hurt? Instead of just checking for file or profile existence, I guess we try to validate if they can actually be used to retrieve creds.

Reply:
Not necessarily, but it can lead to different behavior. The original implementation meant that if we had a `Profile` or `Filename` set, then it would only try the SharedCredentialsProvider even if the properties were configured incorrectly. Now, we'll fallback on the SDK chain.

Reply:
I do not think that behavior can be upstreamed.

Reply:
I think falling back if it can't get creds is fine even if the current agent does not.